### PR TITLE
[th2-2150] toProto by parentEventId/bookName/(bookName + scope)

### PR DESCRIPTION
**Pull request by andreydrobynin**

### CODE REVIEW

File: src/main/java/com/exactpro/th2/common/event/Event.java [hunk 1]
```diff
@@ -251,11 +240,84 @@ public Event messageID(MessageID attachedMessageID) {
         return this;
     }
 
-    public List<com.exactpro.th2.common.grpc.Event> toListProto(@Nullable EventID parentID) throws JsonProcessingException {
-        return collectSubEvents(new ArrayList<>(), parentID);
+    public List<com.exactpro.th2.common.grpc.Event> toListProto(@NotNull EventID parentId) throws JsonProcessingException {
```
Comment (Nikita-Smirnov-Exactpro) [inline src/main/java/com/exactpro/th2/common/event/Event.java:243]: Please apply the requiredNotNull method for all NotNull variable

Comment (Nikita-Smirnov-Exactpro) [reply src/main/java/com/exactpro/th2/common/event/Event.java:243]: Also, string values should not be blunk

File: src/main/java/com/exactpro/th2/common/schema/factory/AbstractCommonFactory.java [hunk 2]
```diff
@@ -642,13 +642,12 @@ public EventID getRootEventId() {
                     }
```
Comment (Nikita-Smirnov-Exactpro) [inline src/main/java/com/exactpro/th2/common/schema/factory/AbstractCommonFactory.java:642]: the getRootEventId method must not return null

File: src/main/kotlin/com/exactpro/th2/common/schema/message/impl/monitor/EventMessageRouterMonitor.kt [hunk 3]
```diff
@@ -25,8 +25,8 @@ import org.slf4j.helpers.MessageFormatter.arrayFormat
 
 class EventMessageRouterMonitor(
     private val router: MessageRouter<EventBatch>,
-    private val parentEventID: EventID?,
-    private val bookName: String
+    private val parentEventId: EventID?,
```
Comment (Nikita-Smirnov-Exactpro) [inline src/main/kotlin/com/exactpro/th2/common/schema/message/impl/monitor/EventMessageRouterMonitor.kt:28]:
```suggestion
    private val parentEventId: EventID,
```

File: src/test/java/com/exactpro/th2/common/event/bean/TreeTableTest.java [hunk 4]
```diff
@@ -40,8 +40,10 @@ public void testSerializationRow() throws IOException {
 
         TreeTableBuilder treeTableBuilder = new TreeTableBuilder();
         TreeTable treeTable = treeTableBuilder.row("FirstRow", row).build();
-        com.exactpro.th2.common.grpc.Event event =
-                Event.start().bodyData(treeTable).bookName(BOOK_NAME).toProto(toEventID(BOOK_NAME, "id"));
+        com.exactpro.th2.common.grpc.Event event = Event
+                .start()
+                .bodyData(treeTable)
+                .toProto(toEventID(BOOK_NAME, "id"));
```
Comment (Nikita-Smirnov-Exactpro) [inline src/test/java/com/exactpro/th2/common/event/bean/TreeTableTest.java:46]: Maybe extract the `toEventID(BOOK_NAME, "id")` to a constant value

**Review by Nikita-Smirnov-Exactpro — CHANGES_REQUESTED** [review]

File: src/main/java/com/exactpro/th2/common/event/Event.java (hunk 1 above)
Comment (andreydrobynin) [reply src/main/java/com/exactpro/th2/common/event/Event.java:243]: done

File: src/main/java/com/exactpro/th2/common/schema/factory/AbstractCommonFactory.java (hunk 2 above)
Comment (andreydrobynin) [reply src/main/java/com/exactpro/th2/common/schema/factory/AbstractCommonFactory.java]: done

File: src/test/java/com/exactpro/th2/common/event/bean/TreeTableTest.java (hunk 4 above)
Comment (andreydrobynin) [reply src/test/java/com/exactpro/th2/common/event/bean/TreeTableTest.java]: done

File: src/main/java/com/exactpro/th2/common/schema/box/configuration/BoxConfiguration.java [hunk 5]
```diff
@@ -38,11 +41,15 @@ public void setBoxName(@Nullable String boxName) {
...
+    @NotNull
     public String getBookName() {
         return bookName;
     }
 
-    public void setBookName(String bookName) {
+    public void setBookName(@NotNull String bookName) {
+        if (isBlank(bookName)) {
+            throw new IllegalArgumentException("Book name cannot be null or blank");
+        }
```
Comment (Nikita-Smirnov-Exactpro) [inline src/main/java/com/exactpro/th2/common/schema/box/configuration/BoxConfiguration.java:52]: Use the requireNonBlankBookName method

**Review by Nikita-Smirnov-Exactpro — APPROVED** [review]

File: src/main/java/com/exactpro/th2/common/schema/box/configuration/BoxConfiguration.java (hunk 5 above)
Comment (andreydrobynin) [reply src/main/java/com/exactpro/th2/common/schema/box/configuration/BoxConfiguration.java]: Thanks, I missed it.
Also I change `requireNonNullParentId()`/`requireNonBlankBookName()`/`requireNonBlankScope()` to public

**Review by Nikita-Smirnov-Exactpro — APPROVED** [review]

**Pull request merged**